vendor/google.golang.org/protobuf/encoding/protojson/well_known_types.go

```diff
@@ -814,16 +814,22 @@ func (d decoder) unmarshalTimestamp(m protoreflect.Message) error {
 		return d.unexpectedTokenError(tok)
 	}
 
-	t, err := time.Parse(time.RFC3339Nano, tok.ParsedString())
+	s := tok.ParsedString()
+	t, err := time.Parse(time.RFC3339Nano, s)
 	if err != nil {
 		return d.newError(tok.Pos(), "invalid %v value %v", genid.Timestamp_message_fullname, tok.RawString())
 	}
-	// Validate seconds. No need to validate nanos because time.Parse would have
-	// covered that already.
+	// Validate seconds.
 	secs := t.Unix()
 	if secs < minTimestampSeconds || secs > maxTimestampSeconds {
 		return d.newError(tok.Pos(), "%v value out of range: %v", genid.Timestamp_message_fullname, tok.RawString())
 	}
+	// Validate subseconds.
```

Comment:
@jpbetz it looks like cel-go makes use of protojson.Marshal / Unmarshal ... does our use interact with timestamps in a way that would be affected by this?

Reply:
This looks like a covers a very specific edge case where the length of the subsections part is too long? 

We test happy path behavior with https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go#L267 but don't test for this edge case. 

At a glance, it appears that this tightening of validation could break CEL expressions, but TBH, I would be shocked if this impacts any actual use of CEL in k8s since data literals of date times are rare.

cc @TristonianJones for broader CEL ecosystem impact.

Reply:
it looks like this was adjusting to keep identical behavior under go1.20, which included http://go.dev/cl/425037